docs/source/blocks/extensions.md

```diff
@@ -1,10 +1,10 @@
 ---
 myst:
   html_meta:
-    "description": "A common pattern in blocks is the 'variations' pattern - a slightly different versions of a block that can be toggled on demand by the editors."
-    "property=og:description": "A common pattern in blocks is the 'variations' pattern - a slightly different versions of a block that can be toggled on demand by the editors."
-    "property=og:title": "Block extensions mechanism"
-    "keywords": "Volto, Plone, frontend, React, Upgrade, Guide, Block extensions, variations, schema enhancers"
+    'description': "A common pattern in blocks is the 'variations' pattern - a slightly different versions of a block that can be toggled on demand by the editors."
+    'property=og:description': "A common pattern in blocks is the 'variations' pattern - a slightly different versions of a block that can be toggled on demand by the editors."
+    'property=og:title': 'Block extensions mechanism'
+    'keywords': 'Volto, Plone, frontend, React, Upgrade, Guide, Block extensions, variations, schema enhancers'
```

Comment:
@stevepiercy The conversion to single quotes is done by `prettier`, because of `"singleQuotes": true` in `.prettierrc`. There's nothing editor-specific here, except that VS Code runs prettier on save.

I know you hate them here, so I looked for a way to use double quotes in the YAML header of Markdown files but single quotes in `jsx` code blocks. I couldn't find a way to do that; prettier has a way to override this setting, but it's based on file extension rather than syntax. I think it's more important to have single quotes in code samples than double quotes in the YAML header, so I'm leaving it as is for now. :-/

Reply:
I'm sorry, but this makes this file inconsistent with all other documentation files. Perhaps add `*.md` to `.prettierignore`?

It is also possible to add code samples as separate files, run prettier on them for their specific language, and use `literalinclude` to pull them in. I wonder if that would be easier to maintain in the long-term? It is a couple more steps for the author, but then we would get consistently formatted code samples which benefits everyone, especially the reader.

Reply:
As I think about it more, this would have an additional benefit. Docs uses Pygments for syntax highlighting and somewhat for checking. It indicates where the problem is (sometimes off by a line), but it doesn't explain what is the problem. For example:

```console
/Users/stevepiercy/projects/Plone/documentation/submodules/volto/docs/source/addons/theme.md:41: WARNING: Could not lex literal_block as "less". Highlighting skipped.
/Users/stevepiercy/projects/Plone/documentation/submodules/volto/docs/source/theming/custom-styling.md:104: WARNING: Could not lex literal_block as "less". Highlighting skipped.
```

Then we could run linters on these separate files and get better error messages. If it is useful, I could add it to the Makefile and CI.

Reply:
Thanks for the reversion and consideration.

Checking the validity of code in docs is a challenge, specifically code snippets, not shell commands and console output. Some packages have good solutions, but they are mostly centered on APIs or UI widgets, like our REST API and Storybook. Some tutorials use `literalinclude` well, such as Pyramid's [SQLAlchemy + URL dispatch wiki tutorial](https://docs.pylonsproject.org/projects/pyramid/en/latest/tutorials/wiki2/basiclayout.html), but that is a tutorial to build a sample app and has tests that can be run. I'll ask amongst the documentarian community to see what they do for code snippets.

Reply:
@stevepiercy In the meantime, I would approve your suggestion to disable prettier for .md files. It'll mean we don't get any auto-formatting of code blocks, but at least we won't get accidental drive-by reformattings when people are just trying to add a sentence or two.

Reply:
@davisagli let's do it! With Sphinx and Pygments, the code samples are validated well enough.